core/src/main/java/com/scalar/db/storage/cosmos/CosmosAdmin.java

```diff
@@ -466,25 +484,29 @@ private ThroughputProperties calculateThroughput(Map<String, String> options) {
 
   @Override
   public boolean namespaceExists(String namespace) throws ExecutionException {
-    return databaseExists(namespace);
+    try {
+      getNamespacesContainer()
+          .readItem(namespace, new PartitionKey(namespace), CosmosNamespace.class);
+      return true;
+    } catch (RuntimeException e) {
+      if (e instanceof CosmosException
+          && ((CosmosException) e).getStatusCode() == CosmosErrorCode.NOT_FOUND.get()) {
+        return false;
+      }
+      throw new ExecutionException("Checking if the namespace exists failed", e);
+    }
   }
```

Comment:
For each admin, the implementation of `admin.namespaceExists()` is now uniformized and checks if the namespace is contained in the namespace metadata table.